exchanger/forwarder.go

```diff
@@ -0,0 +1,49 @@
+package exchanger
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/miekg/dns"
+)
+
+// A Forwarder is a DNS message forwarder that transparently proxies messages
+// to DNS servers.
+type Forwarder func(*dns.Msg, string) (*dns.Msg, error)
+
+// Forward is an utility method that calls f itself.
+func (f Forwarder) Forward(m *dns.Msg, proto string) (*dns.Msg, error) {
```

Comment:
There is no need, that's why it's an utility function. Code reads much better with it. If we define an interface, it would also implement it automatically.